app/views/bookings/show.html.erb

```diff
@@ -125,6 +125,20 @@
                             <div>Total (A+B+C+D) = <%= @booking.cost %> </div>
                         </td>
                     </tr>
+                    <% elsif @booking.is_cancelled? %>
```

Comment:
we show cost to only admin and owner so you have to add an and condition similar to above if condition. 
`and current_user.is_admin_or_owner?`